overlord/devicestate/devicestate.go

```diff
@@ -1415,6 +1415,11 @@ func Remodel(st *state.State, new *asserts.Model, localSnaps []*snap.SideInfo, p
 		}
 	}
 
+	// check for exclusive changes again since we released the lock
```

Comment:
the comment seems misplaces, AFAIU lock was not released here at any point

Reply:
We release the lock [here](https://github.com/snapcore/snapd/blob/master/overlord/devicestate/devicestate.go?plain=1#L1384) and in `remodelTasks` when we talk to the store to [fetch validation sets](https://github.com/snapcore/snapd/blob/master/overlord/assertstate/assertstate.go?plain=1#L1171). A similar explanation is given when we check that the model hasn't changed from under us: https://github.com/snapcore/snapd/blob/master/overlord/devicestate/devicestate.go?plain=1#L1399